packages/appsync-modelgen-plugin/src/utils/process-connections.ts

```diff
@@ -17,14 +17,14 @@ export type CodeGenFieldConnectionBelongsTo = CodeGenConnectionTypeBase & {
   kind: CodeGenConnectionType.BELONGS_TO;
   targetName: string; // Legacy field remained for backward compatability
   targetNames: string[]; // New attribute for v2 custom pk support
-
+  reference?: boolean;
```

Comment:
see naming nit above.

Do we need to modify the publicly exported type? If so, we need to run this change by studio, console, runtime folks to make sure the new field isn't going to break anybody. It seems like the only reason you need it is for internal processing, so using a dedicated internal type would prevent this implementation detail from leaking into your public abstraction.

If we for some reason do need the consumers to know about references, then that's a whole other conversation we need to have. :)

Reply:
These types are for the intermediate representation and are not exported outside of the package. i.e. No codegen outputs will contain this field.

This boolean is only necessary for this case: https://github.com/aws-amplify/amplify-codegen/blob/b2780a920ca7295a3098bda4c7ed50be7686d9ea/packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts#L1193-L1212

It would be possible to re-process the relationship from here, but I think it would become much more complicated.